lib/src/realm_class.dart

```diff
@@ -173,31 +173,41 @@ class Realm implements Finalizable {
     if (cancellationToken != null && cancellationToken.isCancelled) {
       throw cancellationToken.exception!;
     }
-    final realm = Realm(config);
-    StreamSubscription<SyncProgress>? subscription;
-    try {
-      if (config is FlexibleSyncConfiguration) {
-        final session = realm.syncSession;
-        if (onProgressCallback != null) {
-          subscription = session.getProgressStream(ProgressDirection.download, ProgressMode.forCurrentlyOutstandingWork).listen(onProgressCallback);
-        }
-        await session.waitForDownload(cancellationToken);
-        await subscription?.cancel();
-      }
-    } catch (_) {
-      await subscription?.cancel();
-      realm.close();
-      rethrow;
+
+    if (config is! FlexibleSyncConfiguration) {
+      final realm = Realm(config);
+      return await CancellableFuture.value(realm, cancellationToken);
     }
-    return await CancellableFuture.value(realm, cancellationToken);
+
+    _ensureDirectory(config);
+
+    final asyncOpenHandle = realmCore.createRealmAsyncOpenTask(config);
+
+    return CancellableFuture.from<Realm>(() async {
+      StreamSubscription<SyncProgress>? progressSubscription;
+      if (onProgressCallback != null) {
+        final progressController = RealmAsyncOpenProgressNotificationsController._(asyncOpenHandle);
+        final progressStream = progressController.createStream();
+        progressSubscription = progressStream.listen(onProgressCallback);
```

Comment:
yeah, we can do it, but that's part of the public interface already and being named parameter it would be a breaking change.  Then the `CancellationToken? cancellationToken` would be just `token`. not sure we gain much.